python/ray/serve/_private/result_wrapper.py

```diff
@@ -0,0 +1,148 @@
+import threading
+import time
+from abc import ABC, abstractmethod
+from typing import Callable, Optional, Union
+
+import ray
+from ray.serve._private.utils import calculate_remaining_timeout
+
+
+class ResultWrapper(ABC):
+    @abstractmethod
+    def get(self, timeout_s: Optional[float]):
+        raise NotImplementedError
+
+    @abstractmethod
+    async def get_async(self):
+        raise NotImplementedError
+
+    @abstractmethod
+    def next(self):
+        raise NotImplementedError
+
+    @abstractmethod
+    async def anext(self):
+        raise NotImplementedError
+
+    @abstractmethod
+    async def add_callback(self, callback: Callable):
```

Comment:
same here does it need to be `async def`?

Reply:
oh maybe because it might resolve something under the hood?

Reply:
this is also typo